connector/src/main/java/org/jboss/as/connector/subsystems/datasources/DataSourceEnable.java

```diff
@@ -78,27 +78,28 @@ public void execute(OperationContext context, ModelNode operation) {
 
         final Resource resource = context.readResourceForUpdate(PathAddress.EMPTY_ADDRESS);
         final ModelNode model = resource.getModel();
-        if (! context.isBooting())
+        if (!context.isBooting())
             model.get(ENABLED).set(true);
 
-        if (context.isNormalServer()) {
-            DataSourceStatisticsService.registerStatisticsResources(resource);
-
+        if (context.isBooting()) {
             context.addStep(new OperationStepHandler() {
                 public void execute(OperationContext context, ModelNode operation) throws OperationFailedException {
                     final ManagementResourceRegistration registration = context.getResourceRegistrationForUpdate();
                     addServices(context, operation, registration, model, isXa());
                     context.completeStep(new OperationContext.RollbackHandler() {
-                                            @Override
-                                            public void handleRollback(OperationContext context, ModelNode operation) {
-                                            }
-                                        });
+                        @Override
+                        public void handleRollback(OperationContext context, ModelNode operation) {
+                        }
+                    });
```

Comment:
L89-93 can just be

```
context.completeStep(OperationContext.RollbackHandler.NOOP_ROLLBACK_HANDLER);
```

Starting with WildFly 9 you can also skip it altogether now. Registering a no-op like OperationContext.RollbackHandler.NOOP_ROLLBACK_HANDLER basically amounts to a comment showing that doing nothing was deliberate.